### PR TITLE
ui: Persist LocalSetting values in sessionStorage

### PR DESCRIPTION
**Pull request by koorosh**
Current LocalSettings implementation does store data in reducer only,
so it won't be persisted after the application reload.
To make local settings persistent during session, it is saved in
sessionStorage.
- Values stored in sessionStorage are prefixed with 'cockroachui' key;
- Stored values are retrieved on demand and used as a fallback in case
The following values are not present in redux store.

For better code organization, `rootSaga` is introduced to combine existing sagas and run them in parallel.

Depends on PR in `yarn-vendor` repo: https://github.com/cockroachdb/yarn-vendored/pull/11

### CODE REVIEW

**Review by dhartunian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @Koorosh)

___
*[pkg/ui/src/redux/alerts.spec.ts, line 37 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/42984#-LwFLQE0-7_Oe7_dTML_:-LwFLQE0-7_Oe7_dTMLa:bi762n) ([raw file](https://github.com/cockroachdb/cockroach/blob/18452be6421456dbf0b37d7435e134866a4c4b50/pkg/ui/src/redux/alerts.spec.ts#L37)):*
> ```XML
> } from "./apiReducers";
> 
> const sandbox = sinon.createSandbox();
> ```

what is the reason to create a sinon sandbox instead of using the default sandbox?

**Comment by Koorosh** [comment]
`sinon.createSandbox ` creates a single context that can be restored all together (all mocked/stubbed functions) instead of restoring every function separately. It might not reasonable for doing that in the current situation because only one function is stubbed, but it is a good option when more `stubs`/`mocks` will be introduced in tests (so for next stubs we should not care about restoring them).
Basically, it reduces the risk of forgetting to restore some mocked functions in default `sinon` instance (which is shared between all running tests) and get into the situation when one test affects other tests.

**Review by koorosh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[pkg/ui/src/redux/alerts.spec.ts, line 37 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/42984#-LwFLQE0-7_Oe7_dTML_:-LwID9ia8O8kIrgt8YGv:b-jvj05e) ([raw file](https://github.com/cockroachdb/cockroach/blob/18452be6421456dbf0b37d7435e134866a4c4b50/pkg/ui/src/redux/alerts.spec.ts#L37)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

what is the reason to create a sinon sandbox instead of using the default sandbox?
</blockquote></details>

`sinon.createSandbox` creates a single context that can be restored all together (all mocked/stubbed functions) instead of restoring every function separately. It might not reasonable for doing that in the current situation because only one function is stubbed, but it is a good option when more stubs/mocks will be introduced in tests (so for next stubs we should not care about restoring them).
Basically, it reduces the risk of forgetting to restore some mocked functions in default `sinon` instance (which is shared between all running tests) and get into the situation when one test affects other tests.

**Review by koorosh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[pkg/ui/src/redux/alerts.spec.ts, line 37 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/42984#-LwFLQE0-7_Oe7_dTML_:-LwIFDRoD1jhfOLpU3Fo:b-qs62ni) ([raw file](https://github.com/cockroachdb/cockroach/blob/18452be6421456dbf0b37d7435e134866a4c4b50/pkg/ui/src/redux/alerts.spec.ts#L37)):*
<details><summary><i>Previously, Koorosh (Andrii Vorobiov) wrote…</i></summary><blockquote>

`sinon.createSandbox` creates a single context that can be restored all together (all mocked/stubbed functions) instead of restoring every function separately. It might not reasonable for doing that in the current situation because only one function is stubbed, but it is a good option when more stubs/mocks will be introduced in tests (so for next stubs we should not care about restoring them).
Basically, it reduces the risk of forgetting to restore some mocked functions in default `sinon` instance (which is shared between all running tests) and get into the situation when one test affects other tests.
</blockquote></details>

One more reason for *not* using `#createSandbox` is a required version upgrade for `sinon` library. 
I considered it is a good reason to gradually upgrade project dependencies as needed.

**Review by dhartunian — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @Koorosh)

___
*[pkg/ui/src/redux/alerts.spec.ts, line 37 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/42984#-LwFLQE0-7_Oe7_dTML_:-LwJcKCG6BT4kZYe7_oA:bnm1sqt) ([raw file](https://github.com/cockroachdb/cockroach/blob/18452be6421456dbf0b37d7435e134866a4c4b50/pkg/ui/src/redux/alerts.spec.ts#L37)):*
<details><summary><i>Previously, Koorosh (Andrii Vorobiov) wrote…</i></summary><blockquote>

One more reason for *not* using `#createSandbox` is a required version upgrade for `sinon` library. 
I considered it is a good reason to gradually upgrade project dependencies as needed.
</blockquote></details>

Thanks for clarifying @koorosh. That was helpful! LGTM
